tests/timeseries/_pv_pool/test_pv_pool_control_methods.py

```diff
@@ -0,0 +1,262 @@
+        await pv_pool.propose_power(Power.from_watts(-4000.0))
+        await self._recv_reports_until(
+            bounds_rx,
+            lambda x: x.target_power is not None
+            and x.target_power.as_watts() == -4000.0,
+        )
+        self._assert_report(
+            await bounds_rx.receive(), power=-4000.0, lower=-100000.0, upper=0.0
+        )
+        await asyncio.sleep(0.0)
+
+        # Components are set initial power
+        assert set_power.call_count == 4
+        inv_ids = mocks.microgrid.pv_inverter_ids
+        assert sorted(set_power.call_args_list, key=lambda x: x.args[0]) == [
+            mocker.call(inv_ids[0], -1000.0),
+            mocker.call(inv_ids[1], -1000.0),
+            mocker.call(inv_ids[2], -1000.0),
+            mocker.call(inv_ids[3], -1000.0),
+        ]
+
+        # After failing 1 inverter, bounds should go down and power shouldn't be
+        # distributed to that inverter.
+        await self._fail_pv_inverters([inv_ids[1]], mocks)
+        await self._recv_reports_until(
+            bounds_rx,
+            lambda x: x.bounds is not None and x.bounds.lower.as_watts() == -80000.0,
+        )
+        self._assert_report(
+            await bounds_rx.receive(), power=-4000.0, lower=-80000.0, upper=0.0
+        )
+
+        set_power.reset_mock()
+        await pv_pool.propose_power(Power.from_watts(-70000.0))
+        await self._recv_reports_until(
+            bounds_rx,
+            lambda x: x.target_power is not None
+            and x.target_power.as_watts() == -70000.0,
+        )
+
+        self._assert_report(
+            await bounds_rx.receive(), power=-70000.0, lower=-80000.0, upper=0.0
+        )
+        await asyncio.sleep(0.0)
+
+        # Components are set initial power
+        assert set_power.call_count == 3
+        inv_ids = mocks.microgrid.pv_inverter_ids
+        assert sorted(set_power.call_args_list, key=lambda x: x.args[0]) == [
+            mocker.call(inv_ids[0], -10000.0),
+            mocker.call(inv_ids[2], -30000.0),
+            mocker.call(inv_ids[3], -30000.0),
+        ]
+
+        # After the failed inverter recovers, bounds should go back up and power
+        # should be distributed to all inverters
+        await self._fail_pv_inverters([], mocks)
+        await self._recv_reports_until(
+            bounds_rx,
+            lambda x: x.bounds is not None and x.bounds.lower.as_watts() == -100000.0,
+        )
+        self._assert_report(
+            await bounds_rx.receive(), power=-70000.0, lower=-100000.0, upper=0.0
+        )
+
+        set_power.reset_mock()
+        await pv_pool.propose_power(Power.from_watts(-90000.0))
+        await self._recv_reports_until(
+            bounds_rx,
+            lambda x: x.target_power is not None
+            and x.target_power.as_watts() == -90000.0,
+        )
+
+        self._assert_report(
+            await bounds_rx.receive(), power=-90000.0, lower=-100000.0, upper=0.0
```

Comment:
No tests setting the bounds so `upper` < 0.0, is that on purpose for some reason?

Reply:
Upper is not expected to be less than zero.  We can set arbitrarily small discharge powers.